### PR TITLE
feat(mobile): actionable needs-input notifications with approve and reply

### PR DESCRIPTION
**Pull request by iscekic**
## Changelog for users
- Needs-input notifications now carry Approve and Reply buttons, so the agent can be answered from the shade.
- A question offers Reply; a permission offers Approve; Open pull request appears only when the session has a pull request.
- Approving answers the waiting agent with the app closed, and the notification then reads "Request approved".
- Replying sends the typed answer with the app closed, and the notification then reads "Reply sent".
- A failed Approve or Reply keeps its buttons and reads "Couldn't answer. Tap again to retry.".
- An action that can no longer be answered reads "This request is no longer waiting." and drops its buttons.
- Open pull request lands on the PR review screen; Open session lands on that session's chat.
- A needs-input notification breaks through as time-sensitive, while ordinary agent progress stays quiet; buttons follow the app language.

## Changelog for maintainers
- Shared `@kilocode/notifications` contract pins the wire ids: categories `kilo-needs-input:<kind>[-pr]`, actions `kilo:approve`, `kilo:reply`, `kilo:open-pr`, `kilo:open-session`. Keep them byte-stable; they are not copy.
- Attention pushes gain optional `attentionKind` and `prUrl` in `data`; the notifications worker sets `categoryId` and `interruptionLevel: 'time-sensitive'` only for `category: 'attention'`, so status, chat, lifecycle, balance, and security messages keep their exact shape.
- Android gains a high-importance `agent-attention` channel and `agent` drops to default for progress; iOS adds the Time Sensitive Notifications entitlement, asserted by `assert:config`.
- The app posts its own raise from the cached live rows (pure `planNeedsInputNotifications`, per-session id `needs-input:<id>`) and suppresses the duplicate server push through the posted marker; one raise must stay on exactly one OS notification.
- Headless Approve and Reply run through `runNeedsInputInteraction`, which builds the same mobile session manager the in-app blocking card uses and calls `respondToPermission` / `answerQuestion`. Do not add a second approval path. Outcomes `ok` / `retryable` / `unavailable` pick the result body.
- The background notification task keeps one shared name, registered from `index.js` (headless entry) and the root layout; its executor lazy-loads `./notifications`. A second name would run an action twice.
- Session ingest holds a CLI-disconnect attention clear for a 10-minute absence window and cancels it when a live CLI re-owns the session, so a transport blip cannot consume a live raise.
- iOS uses the same category contract and entitlement; this host has no iOS simulator, so iOS was not driven on a device. Review the three headless payload shapes in `parseResponseData` and the language re-registration next.

## E2E proof

**[e1] kwf-fix proof-01323f3 — PR 6277 needs-input notification actions (android emulator-5554)** — App-owned raise posted on android (NotificationRecord tag=needs-input:ses_f4c25b... channel=agent-attention importance=4 actions=2) and tapping Approve on the shade changed it to 'Request approved' with the actions dropped (e1-approve.log); tapping Open session landed on that session's chat (e1-opensession.log: 'SCENE e1opensession3 OK' + title/'Message'); channel split agent-attention mImportance=4 vs agent-progress mImportance=3 (e1-progress-channel.log). The inline Reply text entry and Open PR were not driven: the harness refused further scene calls for e1 after 5 (Reply action present on…

 (capture not retained)

<details>
<summary>Owner request</summary>

> # kwf-fix: proof-01323f3
> # kwf-fix-pr: https://github.com/Kilo-Org/cloud/pull/6277
> Prove the behaviour of PR https://github.com/Kilo-Org/cloud/pull/6277 with a live end-to-end run, and make no code change.
> The pull request description carries no evidence: the '## E2E proof' section carries no log excerpt, so nothing shows the change was driven end to end.
> 
> Run the pull request's own scenarios on a device or a simulator, for every platform its diff touches. Capture the decisive log lines always, and a screenshot as well for a user-visible change. Never a recording: they are gone (owner, 2026-09-16).
> Write the evidence under a '## E2E proof' heading in your own pull request description. Never run gh: the driver publishes.
> If a scenario fails, name it and say why, and still change no product code: this section proves what the branch already carries.
> 
> 
> # The pull request description as it stands now
> 
> This is the live body; you never need to fetch it, and you must not edit it. The driver merges what your final summary says into it.
> 
> ```
> ## Changelog for users
> - Needs-input notifications now carry Approve and Reply buttons, so the agent can be answered from the shade.
> - A question offers Reply; a permission offers Approve; Open pull request appears only when the session has a pull request.
> - Approving answers the waiting agent with the app closed, and the notification then reads "Request approved".
> - Replying sends the typed answer with the app closed, and the notification then reads "Reply sent".
> - A failed Approve or Reply keeps its buttons and reads "Couldn't answer. Tap again to retry.".
> - An action that can no longer be answered reads "This request is no longer waiting." and drops its buttons.
> - Open pull request lands on the PR review screen; Open session lands on that session's chat.
> - A needs-input notification breaks through as time-sensitive, while ordinary agent progress stays quiet; buttons follow the app language.
> 
> ## Changelog for maintainers
> - Shared `@kilocode/notifications` contract pins the wire ids: categories `kilo-needs-input:<kind>[-pr]`, actions `kilo:approve`, `kilo:reply`, `kilo:open-pr`, `kilo:open-session`. Keep them byte-stable; they are not copy.
> - Attention pushes gain optional `attentionKind` and `prUrl` in `data`; the notifications worker sets `categoryId` and `interruptionLevel: 'time-sensitive'` only for `category: 'attention'`, so status, chat, lifecycle, balance, and security messages keep their exact shape.
> - Android gains a high-importance `agent-attention` channel and `agent` drops to default for progress; iOS adds the Time Sensitive Notifications entitlement, asserted by `assert:config`.
> - The app posts its own raise from the cached live rows (pure `planNeedsInputNotifications`, per-session id `needs-input:<id>`) and suppresses the duplicate server push through the posted marker; one raise must stay on exactly one OS notification.
> - Headless Approve and Reply run through `runNeedsInputInteraction`, which builds the same mobile session manager the in-app blocking card uses and calls `respondToPermission` / `answerQuestion`. Do not add a second approval path. Outcomes `ok` / `retryable` / `unavailable` pick the result body.
> - The background notification task keeps one shared name, registered from `index.js` (headless entry) and the root layout; its executor lazy-loads `./notifications`. A second name would run an action twice.
> - Session ingest holds a CLI-disconnect attention clear for a 10-minute absence window and cancels it when a live CLI re-owns the session, so a transport blip cannot consume a live raise.
> - iOS uses the same category contract and entitlement; this host has no iOS simulator, so iOS was not driven on a device. Review the three headless payload shapes in `parseResponseData` and the language re-registration next.
> 
> ## E2E proof
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> **[e9] Non-retryable gone, with the required absence of a CTA.** — Seeded raise ses_f50f7360affegReG1yb0I0jSRI (notification posted with actions=3 and "Approve", e9-notif-dump2.txt), answered it via the CLI ingest surface with the app process killed (e9-answer-elsewhere2.log; status busy in e9-db-before-tap.log), then tapped Approve on the stale shade notification (e9-approve-tap.log "TAP OK 'Approve' matches=1"); the shade read 'This request is no longer waiting.' with no action controls (e9-shade-appalive.xml), the needs-input record carries no actions= field and android.text=String (This request is no longer waiting.) (e9-notif-dump4.txt), and the session…
> 
> ![[e9] Non-retryable gone, with the required absence of a CTA. — e2e-mobile-app/e9-shade-before-answer.png](https://github.com/user-attachments/assets/dce70866-36b8-4aa2-b7aa-30951a8a2cfc)
> 
> **[e9] Non-retryable gone, with the required absence of a CTA.**
> 
> ![[e9] Non-retryable gone, with the required absence of a CTA. — e2e-mobile-app/e9-shade-before-approve.png](https://github.com/user-attachments/assets/063e5a16-3e36-41c2-a6e3-2ed5b65fb220)
> 
> **[e10] Wrong-kind action keeps the raise (c2) — needs:seed.** — Unknown action set on the shade (e10-drive-expanded.xml: 'Agent needs input' beside buttons 'Approve','Reply','Open session'); after tapping Reply and submitting text the notification read "Couldn't answer. Tap again to retry." with 'Approve','Reply','Open session' still present and '1 Needs input' (e10-drive-result.xml), so nothing was sent and the permission stayed raised (screenshots e10-shade-before-reply.png / e10-shade-after-reply.png); fixture = app-owned post temporarily forced to omit attentionKind/report the unknown category, reverted, git status --short empty; the trailing 'tapping…
> 
> ![[e10] Wrong-kind action keeps the raise (c2) — needs:seed. — e2e-mobile-app/e10-shade-before-reply.png](https://github.com/user-attachments/assets/d1467052-8fd5-4e21-b96b-0c2b208424c2)
> 
> **[e9] Non-retryable gone, with the required absence of a CTA.**
> 
> ![[e9] Non-retryable gone, with the required absence of a CTA. — e2e-mobile-app/e9-shade-before-answer.png](https://github.com/user-attachments/assets/dce70866-36b8-4aa2-b7aa-30951a8a2cfc)
> 
> **[e9] Non-retryable gone, with the required absence of a CTA.**
> 
> ![[e9] Non-retryable gone, with the required
> 
> <details>
> <summary>Owner request</summary>
> 
> > Surface: the mobile app, `apps/mobile`. Pushes already arrive and already
> > deep-link into the app. The needs-input state already exists.
> > 
> > # What to build
> > 
> > Actions on the notification itself, so the user answers without opening the
> > app: Approve, Reply, Open PR, Open session.
> > 
> > Approve must approve the waiting agent. Reply must send the typed text as the
> > answer. Open PR and Open session must land on the right screen. Approve and
> > Reply must work with the app closed; they do their work in the background and
> > the notification then reflects the result.
> > 
> > A needs-input notification must be time-sensitive, so it breaks through, while
> > ordinary agent progress stays quiet.
> > 
> > Each action must call the same code path the in-app control uses. Do not
> > duplicate the approval logic into the notification handler.
> > 
> > # What proves it
> > 
> > The runbook names the exact adb commands for the shade. Use them.
> > 
> > 1. Post a real needs-input notification from the app, open the shade, tap
> >    Approve, and show the agent is approved. The proof is the changed state
> >    read after the tap, plus a screenshot of the shade before it.
> > 2. Reply from the notification and show the answer reached the session.
> > 3. Open PR and Open session each land on the right screen.
> > 4. An ordinary progress notification is NOT time-sensitive. Show the
> >    difference.
> > 
> > Do not change `apps/mobile/src/**/kiloclaw/**`.
> > 
> > # Both platforms
> > 
> > Build this on iOS and on Android. Use each platform's own mechanism; the
> > behaviour the user sees must match.
> > 
> > Prove it on the platform this request names. If the other platform's surface
> > cannot be driven on the host that runs this request, build it anyway, say so
> > plainly in the pull request, and name exactly what was built but not proved.
> > Do not narrow the work to one platform.
> > 
> > If one platform has no equivalent capability at all, say which part and why.
> 
> </details>
> ```

</details>
## Open findings (not fixed here)
- [e8] Retryable unhappy and retry — needs:fault. With a permission raise posted, `$KILO_WORKFLOW/e2e/fault.sh <worktree> down cloudflare-session-ingest`, then tap "Approve" in the shade: expected "Couldn't answer. Tap again to retry." with the Approve control still on the notification and the raise s
- [e8] $KILO_WORKFLOW/e2e/appium.sh <udid> script /home/igor_kilocode_ai/.local/share/kwf/sections/notification-actions-approve-and-reply-5ebf/e2e-mobile-app/repro-e8.json --out /home/igor_kilocode_ai/.local/share/kwf/sections/notification-actions-approve-and-reply-5ebf/e2e-mobile-app/repro-e8
- [e8] Retryable unhappy and retry: Android emulator-5554: with cloudflare-session-ingest down the shade's Approve gave "Couldn't answer. Tap again to retry." with the Approve control retained (e8-shade-fault.log), but after fault.sh up cloudflare-session-ingest ('port 13300 up pids 902941') two taps 
- [e10] Approve from the shade with the app foregrounded (proof 1)
- [e10] The raise is time-sensitive and ordinary progress is quiet
- [e11] Approve from the shade with the app foregrounded (proof 1)
- [e12] Reply from the shade (proof 2)
- [e13] Approve and Reply with the app CLOSED
- [e14] Open session lands on the right screen (proof 3a)
- [e15] Open pull request lands on the PR review screen
- [e15] Retryable unhappy and retry — needs:fault
- [e16] Non-retryable gone, with the required absence of a CTA

e1-shade-before-approve

### CODE REVIEW

File: apps/mobile/src/lib/active-sessions-live-sync-mount.tsx [hunk 1]
```diff
@@ -129,7 +66,83 @@ function useActiveSessionsLiveSync(): void {
...
+  const recompute = useCallback(() => {
+    const next = queryClient.getQueryData<CachedActiveSessionsData>(queryKey)?.sessions ?? [];
+    const plan = planNeedsInputNotifications({
+      previous: notified.current,
+      next,
+      pathname: pathnameRef.current,
+      appState: AppState.currentState,
+    });
+    const dismissed = new Set(plan.dismiss);
+    notified.current = [
```
[the remark this answers is not in the document]
Comment (iscekic) [reply apps/mobile/src/lib/active-sessions-live-sync-mount.tsx]: bot: Accepted. Pushed 8704856 (kwf `kwf-fix-review-188b`).

- `apps/mobile/src/lib/active-sessions-live-sync-mount.tsx:107` — accepted: a re-published row now replaces its earlier entry before the carried set is rebuilt, so an untouched raise does not publish again on the next recompute and the set cannot grow without bound. A mounted regression test covers the kind flip plus an untouched recompute.

File: apps/mobile/src/lib/notification-actions.ts [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+      return true;
+    }
+    case NEEDS_INPUT_ACTION_IDS.openPr: {
+      if (!isRaiseData(data)) {
+        await dismissResponseNotification(response);
+        return true;
+      }
+      // An unparseable PR URL falls back to the session route: the user still
+      // lands on the raise instead of nowhere.
+      setPendingDeepLink(prPathForData(data) ?? notificationPathForData(data), 'notification');
```
[the remark this answers is not in the document]
Comment (iscekic) [reply apps/mobile/src/lib/notification-actions.ts:261]: bot: Accepted. Pushed 8704856 (kwf `kwf-fix-review-188b`).

- `apps/mobile/src/lib/notification-actions.ts:239` — accepted: all four action ids clear the last notification response before dispatch, so a later cold start cannot re-answer or re-navigate; the old exit-path clear in the open-session branch is gone.

File: packages/notifications/src/push-presentation.ts [hunk 3]
```diff
@@ -21,7 +24,7 @@ export type AndroidNotificationChannelId = (typeof ANDROID_NOTIFICATION_CHANNELS
 export function androidChannelIdForPushData(data: PushData): AndroidNotificationChannelId {
   switch (data.type) {
     case 'cloud_agent_session':
-      return 'agent';
+      return data.category === 'attention' ? 'agent-attention' : 'agent';
```
[the remark this answers is not in the document]
Comment (iscekic) [reply packages/notifications/src/push-presentation.ts]: bot: Accepted. Pushed 8704856 (kwf `kwf-fix-review-188b`).

- `packages/notifications/src/push-presentation.ts:27` — rejected (no_change): an unknown `agent-attention` channel does not drop the push — expo-notifications presents it on its own high-importance fallback channel (BaseNotificationBuilder.kt:64-76, importance at :143) — and the worker attaches a channel id only to tokens whose registration recorded an app version (NotificationChannelDO.ts:288-292). The file is unchanged.

File: packages/notifications/src/push-presentation.ts [hunk 4]
```diff
@@ -7,7 +7,10 @@ import { translatePush } from './i18n';
  * `Notifications.AndroidImportance` on the mobile side.
  */
 export const ANDROID_NOTIFICATION_CHANNELS = [
-  { id: 'agent', name: 'Agent sessions', importance: 'high' },
+  // A needs-input raise is time-sensitive and breaks through on `agent-attention`;
+  // ordinary agent progress stays quiet on `agent`.
+  { id: 'agent-attention', name: 'Agent needs input', importance: 'high' },
+  { id: 'agent', name: 'Agent sessions', importance: 'default' },
```
[the remark this answers is not in the document]
Comment (iscekic) [reply packages/notifications/src/push-presentation.ts]: bot: Accepted. Pushed 8704856 (kwf `kwf-fix-review-188b`).

- `packages/notifications/src/push-presentation.ts:13` — rejected (no_change): Android owns an existing channel's importance, the declaration governs channel creation, and a raise no longer travels on `agent` — it rides the new high-importance channel — so no user-visible behavior depends on the `agent` declaration. The file is unchanged.

File: services/git-token-service/src/github-user-authorization-service.ts [hunk 5]
```diff
@@ -131,7 +135,14 @@ export class GitHubUserAuthorizationService {
 
       const token = this.decryptCredential(authorization, 'access');
       const repoParts = params.githubRepo.split('/');
-      const endpoint = `https://api.github.com/repos/${encodeURIComponent(repoParts[0])}/${encodeURIComponent(repoParts[1])}`;
+      // Same base override the PR-review client honors (apps/web), so a dev
+      // stub serving api.github.com is probed through the stub instead of
+      // rejecting the stub token against real GitHub and revoking the row.
+      const apiBaseUrl = (this.env.GITHUB_API_BASE_URL ?? DEFAULT_GITHUB_API_BASE_URL).replace(
```
[the remark this answers is not in the document]
Comment (iscekic) [reply services/git-token-service/src/github-user-authorization-service.ts]: bot: Accepted. Pushed 8704856 (kwf `kwf-fix-review-188b`).

- `services/git-token-service/src/github-user-authorization-service.ts:141` — accepted: a blank or whitespace-only `GITHUB_API_BASE_URL` now counts as unset, so the dev probe keeps `https://api.github.com` instead of a relative path. The new test asserts the api.github.com request.

**Comment by iscekic** [comment]
bot: Cross-platform fix. Pushed 62b3232 (kwf `kwf-fix-platform-9001`).

What changed:
- The shared `@kilocode/notifications` contract pins the wire ids: categories `kilo-needs-input:<kind>[-pr]` and actions `kilo:approve`, `kilo:reply`, `kilo:open-pr`, `kilo:open-session`. Keep them byte-stable; they are not copy.
- Attention pushes gain optional `attentionKind` and `prUrl`; the worker marks only `category: 'attention'` with `categoryId` and `interruptionLevel: 'time-sensitive'`, so status, chat, lifecycle, balance, and security messages keep their shape.
- Platform gates are capability gates only: both channel writers no-op on iOS, named in the comment, and Android gains a high-importance `agent-attention` channel while progress drops to default; iOS adds the Time Sensitive Notifications entitlement, asserted by `assert:config`. iOS was built but not driven on a device because this host has no iOS simulator, so the same shared implementation was proved on Android only.
- The app posts its own raise from cached live rows through the pure `planNeedsInputNotifications` (per-session id `needs-input:<id>`); a posted marker suppresses the duplicate server push, and one raise must stay on exactly one OS notification.
- Headless Approve and Reply run through `runNeedsInputInteraction`, which builds the same mobile session manager the in-app blocking card uses and calls `respondToPermission` / `answerQuestion`. Do not add a second approval path; outcomes `ok` / `retryable` / `unavailable` pick the result body.
- The headless answer reads the active-user id through the app's single cross-platform SecureStore entry point, not a direct `expo-secure-store` import, so it resolves the same account the in-app control would. The notification platform-parity test fails any `Platform.OS`/`Platform.select` branch or platform-suffixed import in the shared notification modules.
- The background notification task keeps one shared name, registered from `index.js` (headless entry) and the root layout; its executor lazy-loads `./notifications`. A second name would run an action twice.
- Session ingest holds a CLI-disconnect attention clear for a 10-minute absence window and cancels it when a live CLI re-owns the session, so a transport blip cannot consume a live raise.

File: apps/mobile/src/lib/needs-input-notification.ts [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+
+/**
+ * Post one raise. A rejected schedule (the permission was revoked after login)
+ * is reported and swallowed: the app-owned carrier must never crash the mount
+ * that owns it.
+ */
+async function publishRow(row: NeedsInputNotificationRow): Promise<void> {
+  const data = pushDataForRow(row);
+  try {
+    await Notifications.scheduleNotificationAsync({
```
[the remark this answers is not in the document]
Comment (iscekic) [reply apps/mobile/src/lib/needs-input-notification.ts:308]: bot: Rejected, no code change (kwf `kwf-fix-review-f1d1`).

Why: the finding does not reproduce in this tree

File: services/session-ingest/src/dos/UserConnectionDO.ts [hunk 7]
```diff
@@ -2066,6 +2198,7 @@ export class UserConnectionDO extends DurableObject<Env> {
     if (
       this.lastHeartbeatAt.size === 0 &&
       this.readyPushFireAt.size === 0 &&
+      this.pendingAttentionResetAt.size === 0 &&
```
[the remark this answers is not in the document]
Comment (iscekic) [reply services/session-ingest/src/dos/UserConnectionDO.ts]: bot: Rejected, no code change (kwf `kwf-fix-review-f1d1`).

Why: the finding does not reproduce in this tree

**Comment by iscekic** [comment]
This description names a scenario the proof did not capture:

- not proved live: Empty answer does not consume the raise (c2). With a question raise posted, tap "Reply" in the shade, send nothing (`adb shell input keyevent KEYCODE_ENTER` on the empty RemoteInput, or tap "Send" without typing): expected the notification keeps its Reply/Approve controls and the raise is still waiting — it must NOT read "This request is no longer waiting." Then tap "Reply" again, `adb shell input text 'ship%sit'`, "Send": expected "Reply sent" and the answer in the session. (no capture cited it)

A repeated proof run rebuilds the same evidence, so no proof run is dispatched for a named gap. Merging with this gap open is your decision.

File: packages/notifications/src/push-presentation.ts [hunk 8]
```diff
@@ -21,7 +27,7 @@ export type AndroidNotificationChannelId = (typeof ANDROID_NOTIFICATION_CHANNELS
 export function androidChannelIdForPushData(data: PushData): AndroidNotificationChannelId {
   switch (data.type) {
     case 'cloud_agent_session':
-      return 'agent';
+      return data.category === 'attention' ? 'agent-attention' : 'agent-progress';
```
Comment (pandemicsyn) [inline packages/notifications/src/push-presentation.ts:103]: bot: New Android channels are sent to already-registered legacy clients that have never created them.

Suggested fix: Gate `agent-attention`/`agent-progress` on the first mobile version that creates both channels (or retain the legacy `agent` channel for older versions). `NotificationChannelDO` currently sends any selected `channelId` whenever `app_version` is non-null, while token registration records `nativeApplicationVersion`; therefore an existing pre-release Android installation normally has a non-null version and receives the new ID even though only the updated app creates it. Android 8+ can drop notifications for an explicitly nonexistent channel. Add dispatch tests for a non-null legacy version and both attention and progress routing.

File: services/session-ingest/src/dos/UserConnectionDO.ts [hunk 9]
```diff
@@ -649,6 +699,9 @@ export class UserConnectionDO extends DurableObject<Env> {
         this.scheduleSessionReadyPush(attachment.kiloUserId, session.id, session.title);
       }
       this.sessionOwners.set(session.id, connectionId);
+      // A live CLI owns (or re-owns) the session: whatever raise it was waiting
+      // on is answerable again, so the absence window is over.
+      this.cancelPendingAttentionReset(session.id);
```
Comment (pandemicsyn) [inline services/session-ingest/src/dos/UserConnectionDO.ts:704]: bot: A CLI reconnect after Durable Object eviction can fail to cancel the persisted delayed attention clear.

Suggested fix: Make cancellation delete `attentionReset:<sessionId>` durably even if `pendingAttentionResetAt` has no entry, or await/rebuild the durable reset state before processing reconnect ownership. After eviction, `scheduleNextAlarm()` starts an asynchronous storage-list rebuild, but this reconnect path calls `cancelPendingAttentionReset` before that callback fills the empty mirror. Its early return leaves the KV entry intact; the callback can then repopulate and rearm it, allowing a later alarm to clear an active raise. Add a test that evicts after persisting a hold, reconnects a CLI while mirror rebuild is pending, and verifies the KV key/alarm are removed.

File: apps/mobile/src/lib/active-sessions-live-sync-mount.tsx [hunk 10]
```diff
@@ -129,7 +67,120 @@ function useActiveSessionsLiveSync(): void {
...
+      ),
+    });
+    const dismissed = new Set(plan.dismiss);
+    // A re-published row REPLACES its previous entry instead of joining it. An
+    // append would leave the stale row first in the set, so the next plan's
+    // `find` would compare the raise against the shape it had before the
+    // re-publish, publish again on every recompute, and grow the set without
+    // bound.
+    const republished = new Set(plan.publish.map(row => row.sessionId));
+    notified.current = [
```
Comment (pandemicsyn) [inline apps/mobile/src/lib/active-sessions-live-sync-mount.tsx]: bot: Failed native notification posts and dismissals are treated as completed, so active raises are not retried and stale notifications may persist.

Suggested fix: Commit `notified.current` only for operations that actually succeed, or have `applyNeedsInputNotifications` return failed publish/dismiss identifiers and retain them for a later recompute/retry. This block removes dismissed rows and adds published rows before calling the applier, while `publishRow` and `dismissIdentifier` swallow native failures. A rejected post leaves an active row remembered as posted, so unchanged cache updates plan no retry; a rejected dismissal similarly removes the only remembered identifier, so the stale OS notification is not retried. Add rejection tests for both native methods.

**Review by pandemicsyn — APPROVED** [review]

File: packages/notifications/src/push-presentation.ts (hunk 8 above)
Comment (iscekic) [reply packages/notifications/src/push-presentation.ts:103]: bot: Accepted. Pushed b5374fa (kwf `kwf-fix-review-15f7`).

- `packages/notifications/src/push-presentation.ts:30` — accepted. A server push now resolves the Android channel against the token's registered `app_version`: agent raises and progress reach `agent-attention` or `agent-progress` only from the split build, and older or unknown versions fall back to the legacy `agent` channel those installs created. Non-agent channels are unchanged.

File: services/session-ingest/src/dos/UserConnectionDO.ts (hunk 9 above)
Comment (iscekic) [reply services/session-ingest/src/dos/UserConnectionDO.ts:704]: bot: Accepted. Pushed b5374fa (kwf `kwf-fix-review-15f7`).

- `services/session-ingest/src/dos/UserConnectionDO.ts:704` — accepted. `cancelPendingAttentionReset` now deletes the durable `attentionReset:<sessionId>` entry unconditionally instead of returning early when the in-memory mirror is empty, so a reconnect after eviction cannot leave a hold that the later storage re-list would re-arm.

File: apps/mobile/src/lib/active-sessions-live-sync-mount.tsx (hunk 10 above)
Comment (iscekic) [reply apps/mobile/src/lib/active-sessions-live-sync-mount.tsx]: bot: Accepted. Pushed b5374fa (kwf `kwf-fix-review-15f7`).

- `apps/mobile/src/lib/active-sessions-live-sync-mount.tsx:141` — accepted. `applyNeedsInputNotifications` now reports which posts and dismissals reached the OS, and `reconcileNotifiedAfterApply` restores the rows a failed post or dismissal removed, so a later recompute retries them instead of treating the failed operation as done.
